Documented gymbal lock cases

evbernardes opened the pull request:
#### References to other Issues or PRs
Follow up to #24604 and #24605.

#### Brief description of what is fixed or changed
Modified docstring of `Quaternion.to_euler` to explain how degenerate cases are dealt with.

As another minor edit, replaced `sympify(0)` with `S.Zero`.

#### Release Notes

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

evbernardes committed:
Documented gymbal lock cases

sylee957 commented on sympy/algebras/quaternion.py:
```diff
@@ -442,6 +442,9 @@ def to_euler(self, seq, angle_addition=True, avoid_square_root=False):
         in the sequence given by `seq`. This implements the method described
```
```suggestion
        in the sequence given by ``seq``. This implements the method described
```

evbernardes committed:
minor edit

Co-authored-by: (Lazard) S.Y. Lee  <sylee957@gmail.com>

sylee957 commented:
I recognize that the some math is broken in 
https://output.circle-artifacts.com/output/job/e4dc125b-11b0-4b99-8af4-1d5cf1cd664a/artifacts/0/doc/_build/html/modules/algebras.html#sympy.algebras.Quaternion.to_euler
can you fix the problems with set notation and backticks and the formula in `avoid_square_root`?

evbernardes committed:
Fixed docstring

evbernardes committed:
Merge branch 'document_to_euler_degenerate' of https://github.com/evbernardes/sympy into document_to_euler_degenerate

evbernardes commented:
> I recognize that the some math is broken in https://output.circle-artifacts.com/output/job/e4dc125b-11b0-4b99-8af4-1d5cf1cd664a/artifacts/0/doc/_build/html/modules/algebras.html#sympy.algebras.Quaternion.to_euler can you fix the problems with set notation and backticks and the formula in `avoid_square_root`?

Oops indeed, sorry about that. I think it's fixed now, added double backticks everywhere and fixed the call to the math environment.

evbernardes committed:
Added more double backticks

evbernardes commented:
Ok, I think this is good now. Did a final commit adding double backticks to `from_euler` too.

evbernardes committed:
Minor edit

evbernardes committed:
Changed "Default :" to "Default value:"
